lib/matching/matching.h

```diff
@@ -14,13 +14,17 @@ namespace MouseTrack {
 
 class Matching {
 public:
-  ~Matching(){};
+  virtual ~Matching();
+
   /// Given a list of cluster chains, we find the best matching to append the
   /// given list of descriptors. descriptors: a list of descriptors from a new
   /// frame chains: a list of cluster chains to which we want to appeand one
   /// cluster from descriptors returns a vector of size descriptors.size(),
   /// which holds the index of the matching chain for each descriptor. If no
   /// match was found, the vector value is set to -1.
+  ///
+  /// Note: each cluster chain must have a descriptor for each frame of the
```

Comment:
Nope: each cluster chain must have at least one descriptor and cluster pair belonging to the same frame.

Cluster chains are allowed to have holes. Forbidding holes, would be far to brittle. Just think about all the noise in the data. The clustering could create 10 clusters for frame `i` but 9 cluster for frame `i+1`.

(also see further below the more detailed considerations)